### PR TITLE
SUS-1777: Improve lookup of parent page for Wall/Forum content

### PR DESCRIPTION
**Pull request by mszabo-wikia**
* Change lazy-loading condition in `WallMessage::getArticleTitle` - should fix problem of old cached `WallMessage` instances containing `CommentsIndex` instance instead of `CommentsIndexEntry`
* Skip comments index lookup and build parent title from thread title for Wall threads - because they are known to belong to one user
* `CommentsIndex::entryFromId` now throws exception if lookup fails, for easier handling

https://wikia-inc.atlassian.net/browse/SUS-1777

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WallExternalController.class.php#L77](https://github.com/Wikia/app/blob/e848a3669f617006b60f554276f16de9c8fe7b4d/extensions/wikia/Wall/WallExternalController.class.php#L77): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/Wall/index/CommentsIndex.class.php [hunk 1]
```diff
@@ -167,7 +168,7 @@ public function entryFromId( int $commentId ) {
 				'commentId' => $commentId
 			] );
 
-			return null;
+			throw new CommentsIndexEntryNotFoundException();
```
Comment (macbre) [inline extensions/wikia/Wall/index/CommentsIndex.class.php:171]: :+1:

File: extensions/wikia/Wall/index/CommentsIndexEntryNotFoundException.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ * https://www.gnu.org/copyleft/gpl.html
+ *
+ * @file
+ */
```
Comment (macbre) [inline extensions/wikia/Wall/index/CommentsIndexEntryNotFoundException.php:21]: Do we actually need these comments?

File: extensions/wikia/Wall/WallMessage.class.php [hunk 3]
```diff
@@ -576,46 +582,39 @@ public function getWallPageUrl() {
...
+		// Forum Threads may have been moved to another board - use comments index as data source
+		try {
+			$commentsIndexEntry = $this->getCommentsIndexEntry();
+			$parentPageId = $commentsIndexEntry->getParentPageId();
+
+			$parentTitle =
+				static::$titleCache[$parentPageId] ??
+				Title::newFromID( $parentPageId ) ??
+				Title::newFromID( $parentPageId, Title::GAID_FOR_UPDATE ) ??
+				static::$emptyTitle;
```
Comment (macbre) [inline extensions/wikia/Wall/WallMessage.class.php]: ` static::emptyTitle()` instead?

**Review by macbre — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WallExternalController.class.php#L77](https://github.com/Wikia/app/blob/937858f6b4f3f244fc575a16c9bfbd43c2be5014/extensions/wikia/Wall/WallExternalController.class.php#L77): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WallExternalController.class.php#L77](https://github.com/Wikia/app/blob/405e4624155c33696825f7231e6edda5e4ab19fa/extensions/wikia/Wall/WallExternalController.class.php#L77): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WallExternalController.class.php#L77](https://github.com/Wikia/app/blob/0ec2c070daf632e490433b08eb9644748f44ddad/extensions/wikia/Wall/WallExternalController.class.php#L77): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WallExternalController.class.php#L77](https://github.com/Wikia/app/blob/cc9ccf6469c49cec6ac490f5e27f5c3b340bf608/extensions/wikia/Wall/WallExternalController.class.php#L77): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WallExternalController.class.php#L77](https://github.com/Wikia/app/blob/01bdf0957fddfd40d392c419e5f987a5a6277148/extensions/wikia/Wall/WallExternalController.class.php#L77): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**